Send errors async and shutdown afterwards

mmcdaris opened the pull request:
In some cases the program exits before the error can be sent. This change ensures the error is sent asynchronously and shuts down the executor when the thread exits.

mmcdaris committed:
Send errors async and shutdown afterwards

In some cases the program exits before the error can be sent. This
ensures the error is sent asynchronously and shuts down the executor
when the thread exits.

mmcdaris committed:
add to .gitignore

mmcdaris committed:
Test fix attempts

mmcdaris committed:
Switch to runAsync since we don't need a return

thompiler reviewed (approved):
Ran through this with Morgan and tested out the example repo. Would not hurt to get more eyes on this though (eg: @kyrylo).

Good job investigating this issue @mmcdaris!

kyrylo commented on src/test/java/TestSyncSender.java:
```diff
@@ -0,0 +1,17 @@
+package io.airbrake.log4javabrake2;
```
Does this file really test anything?

mmcdaris replied:
Good point, that's a bit misleading. This file has no tests but provides a kind of mock SyncSender. I'll experiment with renaming it something like `MockSyncSender` instead. Thanks for the review!

mmcdaris committed:
Rename to MockSyncSender for clarity

This class is just a helper so we can inspect the notices that would
have been sent and to avoid actual sending behavior in the tests.
Renaming it makes it more clear that this is not a test.
